docs/DeveloperGuide.md

```diff
@@ -1,29 +1,204 @@
 # Developer Guide
 
 ## Acknowledgements
+Credits to [Personbook](https://github.com/nus-cs2113-AY2223S2/personbook)
+for some reused skeleton code and inspiration on OOP implementation.
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Setting up
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design 
+### Architecture
+![Architecture_Diagram](uml/images/DinerDirectorArchitectureDiagram.png)
+The **Architecture Diagram** given above explains the high-level design of DinerDirector. 
+Given below is a quick overview of main components and how they interact with each other. 
 
+**Architecture of Main Components** <br/>
+`DinerDirector` entry point to the application is DinerDirector. Initially, `DinerDirector` class interact with `UI` class to 
+prompt user for inputs. Once user keys in the input text, the input will be redirected to `Parser` class. `Parser` class will manage the error 
+handling of user's input. Next, any logic will be executed by `Command` object. If there exists any interaction with `Entity`, `Command` will 
+request to `Manager` to assist the CRUD operations.  
+
+### Command Component (Zheng Rong)
+
+##### The command component consists of the following:
```

Comment:
does this need to be a heading? Seems a little unnecessary to me, especially since it is not used in the rest of the document